app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -50,7 +60,39 @@ public Parse(char[] code) throws Exception {
                 }
             }
         }
-        associateIDStruct(comments);
+        parseComments(comments);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    public void decode(int canID, byte[] payload) {
+        String payloadDataType = getConst(canID).payLoadDataType;
+        switch (payloadDataType){
+            case "single":
+                payloadMap.put(canID,
+                        ByteBuffer.wrap(payload).order(ByteOrder.LITTLE_ENDIAN).getFloat());
+                break;
+            case "struct":
+                ArrayList<StructContents> structContents = getStruct(canID);
+                int incr = payload.length / structContents.size();
+                for (int i = 0; i < structContents.size(); i++) {
+                    byte[] byteArray = getBytes(payload, i * incr, incr);
+                    if (incr == 1) {
+                        payloadMap.put(structContents.get(i).name, byteArray[0] & 0xff);
+                    } else {
+                        payloadMap.put(structContents.get(i).name,
+                                ByteBuffer.wrap(byteArray).order(ByteOrder.LITTLE_ENDIAN).getFloat());
+                    }
+                }
+                break;
+        }
+    }
+
+    private byte[] getBytes(byte[] payload, int start, int num) {
```

Comment:
Generally, if some functionality is a basic data structure operation, it's probably part of the standard library and just a StackOverflow / Google search away. I'd include taking a slice of an array into that category. See https://stackoverflow.com/questions/11001720/get-only-part-of-an-array-in-java - reinventing the wheel is not recommended if it can be avoided.